be/src/storage/lake/lake_persistent_index.cpp

```diff
@@ -172,8 +172,7 @@ void LakePersistentIndex::set_difference(KeyIndexSet* key_indexes, const KeyInde
 
 bool LakePersistentIndex::is_memtable_full() const {
     const auto memtable_mem_size = _memtable->memory_usage();
-    // We have two memtable in index, so memtable memory limit means half of `l0_max_mem_usage`.
-    const bool mem_size_exceed = memtable_mem_size >= config::l0_max_mem_usage / 2;
+    const bool mem_size_exceed = memtable_mem_size >= config::l0_max_mem_usage;
```

Comment:
not sure if it is possible to add ut to ensure this behavior, e.g. by checking the file size on disk.

Reply:
> not sure if it is possible to add ut to ensure this behavior, e.g. by checking the file size on disk.

I will add some test cases for these scenarios.